core/chains/evm/txmgr/stuck_tx_detector.go

```diff
@@ -336,6 +342,84 @@ func (d *stuckTxDetector) detectStuckTransactionsScroll(ctx context.Context, txs
 	return stuckTx, nil
 }
 
+// return fraud and overflow transactions
+func (d *stuckTxDetector) detectStuckTransactionsZircuit(ctx context.Context, txs []Tx, blockNum int64) ([]Tx, error) {
+	var err error
+	var fraudTxs, stuckTxs []Tx
+	fraudTxs, err = d.detectFraudTransactionsZircuit(ctx, txs)
+	if err != nil {
+		return txs, err
+	}
+
+	stuckTxs, err = d.detectStuckTransactionsHeuristic(ctx, txs, blockNum)
+	if err != nil {
+		return txs, err
+	}
+
+	// prevent duplicate transactions from the fraudTxs and stuckTxs with a map
+	uniqueTxs := make(map[common.Hash]Tx)
+	for _, tx := range fraudTxs {
+		uniqueTxs[tx.TxAttempts[0].Hash] = tx
+	}
+
+	for _, tx := range stuckTxs {
+		uniqueTxs[tx.TxAttempts[0].Hash] = tx
+	}
```

Comment:
You could also use the tx ID as a unique identifier? Think that might look cleaner to for the map access
```suggestion
	uniqueTxs := make(map[int64]Tx)
	for _, tx := range fraudTxs {
		uniqueTxs[tx.ID] = tx
	}

	for _, tx := range stuckTxs {
		uniqueTxs[tx.ID] = tx
	}
```